src/mmw/js/src/core/views.js

```diff
@@ -79,11 +80,25 @@ var HeaderView = Marionette.ItemView.extend({
     },
 
     templateHelpers: function() {
-        var self = this;
+        var self = this,
+            currentActive = self.appState.get('active_page'),
+            isSelectArea = currentActive === coreUtils.selectAreaPageTitle,
+            isProjects = currentActive === coreUtils.projectsPageTitle,
+            isAnalyze = currentActive === coreUtils.analyzePageTitle,
+            modelPackageNames = _.pluck(settings.get('model_packages'), 'display_name'),
+            isModel = _.contains(modelPackageNames, currentActive),
+            isCompare = currentActive === coreUtils.comparePageTitle,
+            wasAnalyzeVisible = self.appState.get('was_analyze_visible'),
+            wasModelVisible = self.appState.get('was_model_visible'),
+            wasCompareVisible = self.appState.get('was_compare_visible');
```

Comment:
I wonder if it would make sense to initialize some of these variables as null then run everything through a switch/case block or something to set them? I think there are only a few possible states for the app ->

* <img width="247" alt="screen shot 2017-03-01 at 5 55 32 pm" src="https://cloud.githubusercontent.com/assets/4165523/23485186/571b5ff8-fea8-11e6-8cf9-385555de6eba.png">

* <img width="283" alt="screen shot 2017-03-01 at 5 55 41 pm" src="https://cloud.githubusercontent.com/assets/4165523/23485190/5d017060-fea8-11e6-9d6d-6d0515a4a373.png">

* <img width="339" alt="screen shot 2017-03-01 at 5 55 47 pm" src="https://cloud.githubusercontent.com/assets/4165523/23485191/60f23c72-fea8-11e6-8af6-b91b78dcf07c.png">

* <img width="339" alt="screen shot 2017-03-01 at 5 55 52 pm" src="https://cloud.githubusercontent.com/assets/4165523/23485196/655c3c04-fea8-11e6-8fb5-ec84e09b1686.png">

And it seems like those could work as case statements or similar to set these variables?